x-pack/platform/plugins/shared/inference/server/chat_complete/utils/merge_chunks.ts

```diff
@@ -45,7 +45,14 @@ export const mergeChunks = (chunks: ChatCompletionChunkEvent[]): UnvalidatedMess
   );
 
   // some models (Claude not to name it) can have their toolCall index not start at 0, so we remove the null elements
-  message.tool_calls = message.tool_calls.filter((call) => !!call);
+  message.tool_calls = message.tool_calls
+    .filter((call) => !!call)
+    .map((call) => {
+      if (call.function.arguments === '') {
+        call.function.arguments = '{}';
+      }
+      return call;
+    });
```

Comment:
I think it would be better in term of isolation of concerns to do that during the tool call validation step

https://github.com/elastic/kibana/blob/fa1998ce92715f5c7b718ba6972ee8f8938bfa4e/x-pack/platform/plugins/shared/inference/server/util/validate_tool_calls.ts#L43-L45

```diff
-- serializedArguments = JSON.parse(toolCall.function.arguments); 
++ serializedArguments = toolCall.function.arguments ? JSON.parse(toolCall.function.arguments) : {}; 
```

Reply:
see on slack: you can ignore, LGTM